### PR TITLE
Rework interface

### PR DESCRIPTION
**Pull request by aredridel**
* Breaking: make all access go via getter function
* Breaking: Remove all references to the dust bundler
* Feature: Add fastpath support for property access
* Feature: Add a dust formatting function that accepts a model separately from fetching the bundle
* Bug fix: factor out loopalo and use file-resolver more directly
* Bug fix: safe-wrap callbacks, so exceptions thrown by user's callbacks will be diagnosable

### CODE REVIEW

**Comment by aredridel** [comment]
Downsides of this interface: makara 2 can only expose the most raw output without a new method. (`get()` returns a plain bundle (no methods))

File: README.md [hunk 1]
```diff
@@ -40,11 +40,14 @@ var bundle2 = bundalo(config);
 User wants key/values from some bundle file, corrected for locality, and possibly rendered with some data model
 
 ```javascript
-bundle.get({'bundle': 'errors/server','locality': 'en-US', 'model': {'name': 'Will Robinson'}}, function bundaloReturn(err, data) {
-	console.log("what'd we get from bundalo.get?", data, err);
-	cb({
-		'err': data.error
-	});
+bundler.get({bundle: 'errors/server', locality: 'en-US'}, function bundaloReturn(err, bundle) {
```
Comment (grawk) [inline README.md]: returning `bundle` but accessing `data`

Comment (aredridel) [inline README.md]: doh! Incomplete replacement!

**Comment by grawk** [comment]
Re-visiting this. I've been thinking about a rework of the interface as well. But my motivation is wanting to allow the user to supply what we've been calling a "bundler". I.e. what if they want to use something else to process the property strings, like `intl-messageformat` (https://github.com/yahoo/intl-messageformat)?

So we provide a clean interface for plugging that in, while maintaining the resolver/spud/caching features within bundalo. What do you think @aredridel ?

**Comment by grawk** [comment]
I'm trying to see how we can use this PR as a starting point to providing a configurable formatter (i.e. dust or intl-messageformat or something else). It looks like perhaps this wouldn't be too difficult, since the only surface area dust takes up is in the `Bundle.formatDust` method. If instead that was a user-supplied `format` function then we can push the choice of dust (or something else) to the consumer. Does that sound right?

**Comment by grawk** [comment]
The current interface allows you to render multiple messages at once with a data model. That feature is not in this reworked PR. I'm currently looking through usage of bundalo at PayPal and haven't yet come to an instance of a consumer taking advantage of that feature. But wanted to check that it was an intentional omission on your part.

**Comment by aredridel** [comment]
It was indeed: I made the same survey, and found it wanting. If one wants multiple, the caller can combine calls in parallel.

**Comment by aredridel** [comment]
And I like your idea of making `format` generic very much.

**Comment by grawk** [comment]
Another idea is, instead of customizing `format`, allow a custom converter to process the spud output. I've taken a rough stab at this here: https://github.com/krakenjs/bundalo/pull/26

It uses your `simplify-dust` branch as a starting point.

**Comment by grawk** [comment]
Here we go! :+1:

**Pull request merged**